README.md

```diff
@@ -197,25 +197,70 @@ need different templates for different distributions.
 
-If some tasks need to be parameterized according to distribution and version (name of packages,
-configuration file paths, names of services), use this in the beginning of your `tasks/main.yml`:
+You normally use `vars/main.yml` (automatically included) to set variables
+used by your role.  If some variables need to be parameterized according to
+distribution and version (name of packages, configuration file paths, names of
+services), use this in the beginning of your `tasks/main.yml`:
 ```yaml
-- name: Set version specific variables
+- name: Set platform/version specific variables
   include_vars: "{{ item }}"
+  loop:
+    - "{{ role_path }}/vars/{{ ansible_os_family }}.yml"
+    - "{{ role_path }}/vars/{{ ansible_distribution }}.yml"
+    - "{{ role_path }}/vars/{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
+    - "{{ role_path }}/vars/{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
+  when: item is file
+```
+Use `vars/main.yml` for default values for all platforms.  The files in the
+`loop` are in order from least specific to most specific.  Each file in the
+`loop` list will allow you to add or override additional variables to
+specialize the values for platform and/or version.  Using the `when: item is
+file` test means that you do not have to provide all of the `vars/` files,
+only the ones you need.  For example, if every platform except Fedora uses
+`srv_name` for the service name, you can define `myrole_service: srv_name` in
+`vars/main.yml` then define `myrole_service: srv2_name` in `vars/Fedora.yml`.
+In cases where this would lead to duplicate vars files for similiar
+distibutions (e.g. CentOS 7 and RHEL 7), use symlinks to avoid the
+duplication.
+
+Platform specific tasks, however, are different.  You probably want to perform
+platform specific tasks once, for the most specific match.  In that case, use
+`with_first_found` with the file list in order of most specific to least
+specific, including a "default":
+```yaml
+- name: Perform platform/version specific tasks
+  include_tasks: "{{ item }}"
   with_first_found:
-    - "{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
-    - "{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
-    - "{{ ansible_distribution }}.yml"
-    - "{{ ansible_os_family }}.yml"
+    - files:
+        - "setup_{{ ansible_distribution }}_{{ ansible_distribution_version }}.yml"
+        - "setup_{{ ansible_distribution }}_{{ ansible_distribution_major_version }}.yml"
+        - "setup_{{ ansible_distribution }}.yml"
+        - "setup_{{ ansible_os_family }}.yml"
+        - "setup_default.yml"
+      paths:
+        - "{{ role_path }}/tasks"
 ```
+Then you would provide `tasks/setup_default.yml` to do the generic setup, and e.g. `tasks/setup_Fedora.yml` to do the Fedora specific setup.  The "setup_default.yml" is required in order to use `with_first_found`, which will give an error if no file is found.
```

Comment:
This line is too long.

Also, why not suggest using a `setup/` directory instead of forcing tasks to be a flat list of files?

Reply:
fixed in https://github.com/oasis-roles/meta_standards/pull/23/commits/d8266d8254ae62e3e22c4d813a0b5c44d2b6e781